File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.cpp
class Solution {
public:
    int pivotInteger(int n) {
        if (n == 1) return 1;
        for (int i = 1; i < n; i++) {
            // the formula for calculating 1 + 2 + 3 + 4 + .... + n is n * (n + 1) / 2
            int left_side = formula(i);
            int right_side = formula(n) - left_side + i;
            if (left_side == right_side) return i;
        }
        return -1;
    }
    int formula(int n) {
        return n * (n + 1) / 2;
    }
};

File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.rs
impl Solution {
    pub fn pivot_integer(n: i32) -> i32 {
        let s = n * (n + 1) / 2;
        let ans = (s as f64).sqrt() as i32;
        if (ans * ans == s) {
            ans
        } else {
            -1
        }
    }
}